body.tex

```diff
@@ -0,0 +1,442 @@
+
+This will permit in the future to retrieve the information, in order to manually activate the same conda environment before running a build.
+
+The \textbf{rebuild} command could also raise a warning, in case the active environment is different from the one previously used.
+
+
+\subsubsection{New envconfig Command} \label{sec:envconfig}
+
+A new command shall be added, that permits to choose which environment to activate.
+
+It shall accept one of following options:
+
+\begin{itemize}
+\item a \textit{scipipe\_conda\_env} environment reference to activate
+\item an existing build number, in order to activate the environment that has been used for that build.
+\item list the available environments, ready to be activated
+\item compare the active environment with an other one, a \textit{scipipe\_conda\_env} reference or a deployed environment
+\end{itemize}
+
+If no option is provided, the default reference shall be activated, in order to keep the current behavior.
+
+
+\subsection{Changes to lsst/lsst Git Package} \label{sec:newinstall}
+
+The distribution script \textit{newinstall.sh} should be updated in order to be able to deploy and activate the environment corresponding to the EUPS build that is going to be installed.
+
+This imply that the generate scripts \textit{loadLSST.*} shall accept an environment parameter as input, and deploy the environment reference before activating it, if not already deployed.
+
+
+\subsection{Changes to Jenkins builds} \label{sec:cibuilds}
+
+The Jenkins jobs, for example \textit{run-rebuild}, shall be updated in order to accept an environment reference as an optional parameter.
+In this way it will be possible to verify that the science pipelines builds with a new environment.
+
+
+\newpage
+\section{Following activities} \label{sec:follows}
+
+The changes proposed in the previous section \ref{sec:proposed} can be implemented with very minimum impact to the day by day activities.
+
+The new functionalities introduced, can facilitate already the management of the conda environment as it is done now, and the development activities in general.
+
+However, the resolution of the problems exposed in \ref{sec:problems}, requires the reduction of the number of packages included in the shared stack, moving third party libraries in the conda environment.
+
+
+\subsection{Conda Packaging 3rd Party Libraries} \label{sec:3rdpkgs}
+
+Some of the packages listed in appendix \ref{sec:tomove} are already available in Conda public channels or \textit{PyPi} (see following \ref{sec:pypi}).
+They are now included in the Science Pipelines stack as EUPS packages since when they were initially needed, they were not available in any Conda channels.
+
+Some other packages require to be updated more often, therefore they have been included as EUPS packages, since the Conda environment is not sufficiently dynamic.
+
+Packages that requires customization, need to remain part of the stack, at least in a first phase.
```

Comment:
Make clear that "customization" here is packages with substantial forks of the upstream, not packages with minor patches that can be handled by a new recipe.